### PR TITLE
[Resources.Azure] NugetAudit - fix dependencies with known vulnerabilities

### PR DESCRIPTION
**Pull request by Kielek**
Follow up to #2034

## Changes

``` 
    <!-- System.Text.Encodings.Web is indirect reference. It is needed to upgrade it directly to avoid https://github.com/advisories/GHSA-ghhp-997w-qr28 -->
    <PackageReference Include="System.Text.Encodings.Web" Version="4.7.2" />
```
Changelog based on https://github.com/open-telemetry/opentelemetry-dotnet/blob/37535a5607ee7e4056c0e274ec01d1e0111a64be/src/OpenTelemetry.Exporter.Console/CHANGELOG.md#L112-L114

## Merge requirement checklist

* [x] [CONTRIBUTING](https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/main/CONTRIBUTING.md) guidelines followed (license requirements, nullable enabled, static analysis, etc.)
* ~~[ ] Unit tests added/updated~~
* [x] Appropriate `CHANGELOG.md` files updated for non-trivial changes
* [x] Changes in public API reviewed (if applicable)

### CODE REVIEW

File: src/OpenTelemetry.Resources.Azure/OpenTelemetry.Resources.Azure.csproj [hunk 1]
```diff
@@ -15,6 +15,8 @@
   <ItemGroup>
     <PackageReference Include="OpenTelemetry" Version="$(OpenTelemetryCoreLatestVersion)" />
     <PackageReference Include="System.Text.Json" Version="4.7.2" />
+    <!-- System.Text.Encodings.Web is indirect reference. It is needed to upgrade it directly to avoid https://github.com/advisories/GHSA-ghhp-997w-qr28 -->
+    <PackageReference Include="System.Text.Encodings.Web" Version="4.7.2" />
```
Comment (rajkumar-rangaraj) [inline src/OpenTelemetry.Resources.Azure/OpenTelemetry.Resources.Azure.csproj:19]: The .NET Standard 2.0 version of `System.Text.Json` has an indirect dependency on `System.Text.Encodings.Web`. It is better to upgrade `System.Text.Json`, as was done in [OpenTelemetry.Exporter.OneCollector](https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/4791960d8c49adb9e5b45b8900010e3337c06591/src/OpenTelemetry.Exporter.OneCollector/OpenTelemetry.Exporter.OneCollector.csproj#L27)

Comment (rajkumar-rangaraj) [reply src/OpenTelemetry.Resources.Azure/OpenTelemetry.Resources.Azure.csproj:19]: We may need to re-evaluate whether this is still a challenge with auto-instrumentation - https://github.com/open-telemetry/opentelemetry-dotnet-contrib/pull/1279#issuecomment-1652182987

Comment (Kielek) [reply src/OpenTelemetry.Resources.Azure/OpenTelemetry.Resources.Azure.csproj:19]: If possible, I would be more conservative with versioning. This solution is applied in Exporter.Console. I do not see any reason we cannot follow this pattern.

Comment (rajkumar-rangaraj) [reply src/OpenTelemetry.Resources.Azure/OpenTelemetry.Resources.Azure.csproj:19]: We will explore it outside of this PR to understand System.Text.Json version could be bumped.

**Review by rajkumar-rangaraj — APPROVED** [review]

**Pull request merged**